runtime/lua/vim/treesitter/query.lua

```diff
@@ -256,39 +236,48 @@ function M.parse_query(lang, query)
   local cached = query_cache[lang][query]
   if cached then
     return cached
-  else
-    local self = setmetatable({}, Query)
-    self.query = vim._ts_parse_query(lang, query)
-    self.info = self.query:inspect()
-    self.captures = self.info.captures
-    query_cache[lang][query] = self
-    return self
   end
+
+  local self = setmetatable({}, Query)
+  self.query = vim._ts_parse_query(lang, query)
+  self.info = self.query:inspect()
+  self.captures = self.info.captures
+  query_cache[lang][query] = self
+  return self
+end
+
+---@param node TSNode
+---@param source integer|string
+---@param metadata TSMetadata
+---@return Range6
+function M.get_range_from_metadata(node, source, metadata)
```

Comment:
Could be named `get_range`. What other kinds of "get range" cases do we anticipate in the future?